java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowUtils.java

```diff
@@ -528,4 +680,39 @@ private static void updateVector(VarBinaryVector varBinaryVector, Blob blob, boo
     updateVector(varBinaryVector, blob != null ? blob.getBinaryStream() : null, isNonNull, rowCount);
   }
 
+  private static void updateVector(
+      ListVector listVector,
+      ResultSet resultSet,
+      int arrayIndex,
+      int rowCount,
+      JdbcToArrowConfig config)
+      throws SQLException, IOException {
+
+    final JdbcFieldInfo fieldInfo = getJdbcFieldInfoForArraySubType(resultSet.getMetaData(), arrayIndex, config);
+    if (fieldInfo == null) {
+      throw new IllegalStateException("Column " + arrayIndex + " is an array of unknown type.");
```

Comment:
this seems more like config wasn't setup correctly so illegal argument exception might be better.